drivers/can/can_mcp2515.c

```diff
@@ -606,6 +594,25 @@ static struct mcp2515_data mcp2515_data_1 = {
 	.filter_usage = 0U,
 };
 
+BUILD_ASSERT_MSG((DT_INST_0_MICROCHIP_MCP2515_SJW >= 1) &&
+	(DT_INST_0_MICROCHIP_MCP2515_SJW <= 4),
+	"1 <= SJW <= 4");
+BUILD_ASSERT_MSG((DT_INST_0_MICROCHIP_MCP2515_PROP_SEG >= 1) &&
+	(DT_INST_0_MICROCHIP_MCP2515_PROP_SEG <= 8),
+	"1 <= PROP <= 8");
+BUILD_ASSERT_MSG((DT_INST_0_MICROCHIP_MCP2515_PHASE_SEG1 >= 1) &&
+	(DT_INST_0_MICROCHIP_MCP2515_PHASE_SEG1 <= 8),
+	"1 <= BS1 <= 8");
+BUILD_ASSERT_MSG((DT_INST_0_MICROCHIP_MCP2515_PHASE_SEG2 >= 2) &&
+	(DT_INST_0_MICROCHIP_MCP2515_PHASE_SEG2 <= 8),
+	"2 <= BS2 <= 8");
+BUILD_ASSERT_MSG((DT_INST_0_MICROCHIP_MCP2515_PROP_SEG +
+	DT_INST_0_MICROCHIP_MCP2515_PHASE_SEG1) >=
+	DT_INST_0_MICROCHIP_MCP2515_PHASE_SEG2,
+	"PROP + BS1 >= BS2");
+BUILD_ASSERT_MSG(DT_INST_0_MICROCHIP_MCP2515_PHASE_SEG2 >
+	DT_INST_0_MICROCHIP_MCP2515_SJW, "BS2 > SJW");
```

Comment:
I'm checking the datasheet and I think the last two build assert requirements:
`PROP + BS1 >= BS2`
`BS2 > SJW`
are guidelines not strict requirements.  Better as warnings in mcp2515_init()?  Sorry I didn't see that before.

Reply:
They are in the datasheet at 5.5 as a requirement, are you sure it's not fixed?